### DISCUSSION TITLE
Unable to associate file extensions with a bdist_msi

### DISCUSSION THREAD

**Discussion by kbasaran** [marcelotduarte/cx_Freeze#2093] [Q&A] [answered]
I have an application which is able to open files from command line. It uses argparse to get the file name.
https://github.com/kbasaran/Linecraft/blob/0.1.2/main.py
```
    parser.add_argument('infile', nargs="?", type=argparse.FileType('r'), action="store",
                        help="Path to a '*.lc' file. This will open a saved state.")
```
I try to associate the application to '*.lc' files in Windows, using a .msi installer.
https://github.com/kbasaran/Linecraft/blob/0.1.2/setup.py
```
bdist_msi_options = {
    "extensions": [{
                    "extension": "lc",
                    "verb": "load state",
                    "executable": "main.exe",
                    }]
                   }
```
Unfortunately, while the executable from the build still opens files passed to it in command line, double clicking on a '*.lc' file does not work. It doesn't even start the application. I look forward to your advice on this.

**Comment by marcelotduarte** [comment]
Please, check this sample: https://github.com/marcelotduarte/cx_Freeze/tree/main/samples/msi_extensions

The verb should be "open", "print", "view", ...

**Comment by kbasaran** [comment]
OK thank you. This way it does start the application.
Yet the filename doesn't seem to get passed to it as an argument. I added a debug line to print the 'args.infile' and it returns None.

**Reply by marcelotduarte** [reply, marked as answer]
In the sample cited, in the README, 
`4. optionally arguments, e.g. "%1" for proper quoting, but any other program arguments can be passed (see sample),`

and in the setup.py has the use of argument, so in your case:
`"argument": '"%1"',`

**Comment by kbasaran** [comment]
Yes, the argument solved my problem. I had followed this sample and assumed argument was not necessary: https://github.com/marcelotduarte/cx_Freeze/blob/main/samples/msi_extensions/setup.py. But I should have tried it anyways. Thank you for your time.
